docs/images.md

```diff
@@ -5,31 +5,157 @@ You create your Docker image and push it to a registry before referring to it in
 
 The `image` property of a container supports the same syntax as the `docker` command does, including private registries and tags.
 
+## Updating Images
+
+The default pull policy is `PullIfNotPresent` which causes the Kubelet to not pull an image if it already exists. If you would like to always force a pull you must set a pull image policy of `PullAlways` or specify a `:latest` tag on your image.
+
 ## Using a Private Registry
+Private registries may require keys to read images from them.
+Credentials can be provided in several ways:
+  - Per-cluster, automatic
+    - all pods can read the project's private registry
+    - only on GCE/GKE
+  - Per-node keys
+    - all pods can read any configured private registries
+  - Pre-pulled images
+    - all pods can use any images cached on a node
+  - Per-pod keys
+    - only pods which provide own keys can access the private registry
+Each option is described in more detail below.
+   
 
 ### Google Container Registry
-Kubernetes has native support for the [Google Container Registry](https://cloud.google.com/tools/container-registry/), when running on Google Compute Engine.  If you are running your cluster on Google Compute Engine or Google Container Engine, simply use the full image name (e.g. gcr.io/my_project/image:tag) and the kubelet will automatically authenticate and pull down your private image.
+
+Kubernetes has native support for the [Google Container
+Registry](https://cloud.google.com/tools/container-registry/), when running on Google Compute
+Engine (GCE).  If you are running your cluster on GCE or Google Container Engine (GKE), simply
+use the full image name (e.g. gcr.io/my_project/image:tag) and the kubelet will automatically
```

Comment:
Explained that is uses the instance's service account, which only has a read scope.